pkgs/os-specific/linux/nixos-rebuild/nixos-rebuild.sh

```diff
@@ -104,7 +104,8 @@ while [ "$#" -gt 0 ]; do
         fi
         if [ "$1" != system ]; then
             profile="/nix/var/nix/profiles/system-profiles/$1"
-            mkdir -p -m 0755 "$(dirname "$profile")"
```

Comment:
Afaiu it only uses `-p` is only to avoid an error if the directory already exists. If we can rely on the parent existing, a different command would be more precise, e.g.
```
[ -d /nix/var/nix/profiles/system-profiles ] || mkdir -m 0755 "$profile"
```

Reply:
I have no idea whether we can rely on the parent existing. I explicitly wrote this PR to make sure to keep the existing semantics.